operator/v1/types_insights.go

```diff
@@ -0,0 +1,86 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// Insights holds cluster-wide information about Insights.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type Insights struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec is the specification of the desired behavior of the Insights
+	// +kubebuilder:validation:Required
+	// +required
+	Spec InsightsSpec `json:"spec"`
+
+	// status is the most recently observed status of the Insights operator
+	// +optional
+	Status InsightsStatus `json:"status"`
+}
+
+type InsightsSpec struct {
+	OperatorSpec `json:",inline"`
+}
+
+type InsightsStatus struct {
+	OperatorStatus `json:",inline"`
+	// GatherStatus provides basic information about the last Insights data gathering.
+	GatherStatus *GatherStatus `json:"gatherStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
```

Comment:
Explain why nil is allowed.  This complicates all consuming code and I think it only provides value when you have not get gathered any information on a cluster.  However, if you haven't gathered anything, couldn't we also represent that as "no problems found" and be just as accurate and simplify all consuming code?

Reply:
Yes, I've updated it a bit.